src/auth-service/config/constants.js

```diff
@@ -16,7 +16,7 @@ const prodConfig = {
 };
 
 const stageConfig = {
-  MONGO_URI: process.env.MONGO_GCE_URI,
+  MONGO_URI: process.env.MONGO_STAGE_URI,
```

Comment:
I already notice this and took care of it by updating the secrets accordingly. @123MwanjeMike 

The challenge with the current approach is that it was hard to differentiate staging and production so it took me a long time to realise that we were using the same variable for staging and production while giving them different values!
The current setup kinda creates a maintenance nightmare.

Happy to hear your additional thoughts.

Reply:
Hi @Baalmart 
The concern I have with it is the redundancy the approach creates.
It also means we are having production secrets added to the staging .env which is security wise not advisable.

Reply:
@123MwanjeMike 
Perhaps we should not have the production one added to the staging ENV. 
Perhaps you have just not realised the maintenance nightmare this current approach is.

Imagine, as a developer, having to update the value of the ENV variable when testing out different environment. Instead of just running different commands.

It makes development more cumbersome and yet the shared security concerns are some debatable.

Happy to jump on a call to showcase the additional complexities the current approach brings.

In summary, what the current approach attempts to do is already taken care of in the config file of the microservice(s). The ENVS are supposed to be different variables for reasons already shared above.

Reply:
Thanks @Baalmart 
I think I now understand the exhaustion this creates during development.

Although, what I was trying to break is the dependency of the codebase to the environment. In that we(would) always have update the code whenever we have a new environment for the api to run in. In other words we would have 5 different environment variables for just the database connection if we had 5 separate environments instead of just the 2 (stage and prod) we currently have.

But we can for now proceed with it and refactor it in the future.